### PR TITLE
fix: adding Image carousel and test cases

### PR DESCRIPTION
**Pull request by anagarwa**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--maruti-playground--buuhuu.hlx.live/draft/anagarwa/image-carousel
- After: https://image-carousel--maruti-playground--buuhuu.hlx.live/draft/anagarwa/image-carousel-3

### CODE REVIEW

**Review by sdmcraft — COMMENTED** [review]
image

File: component-models.json [hunk 1]
```diff
@@ -73,6 +73,13 @@
         "label": "Image",
         "multi": false
       },
+      {
+        "component": "reference",
+        "valueType": "string",
+        "name": "image_mobile",
```
Comment (buuhuu) [inline component-models.json:79]: both the `image` and `image_mobile` should be in the same group by the naming convention:

`image_desktop` and `image_mobile` should be the property names.

**Comment by anagarwa** [comment]
@buuhuu can you please review the model and definition of Image-carousel ?

**Review by buuhuu — APPROVED** [review]

**Comment by anagarwa** [comment]
@sdmcraft @buuhuu  I have added a sample test case.
`npm run test` should run the test.

File: test/blocks/image-carousel/image-carousel.test.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import { readFile } from '@web/test-runner-commands';
+import decorate from '../../../blocks/image-carousel/image-carousel.js';
+
+document.write(await readFile({ path: './image-carousel.plain.html' }));
+
+const block = document.querySelector('.image-carousel');
+await decorate(block);
+
+describe('Image Carousel Block', () => {
+  it('Test Picture Tag Count', async () => {
```
Comment (buuhuu) [inline test/blocks/image-carousel/image-carousel.test.js:15]: should be phrased like

`it('should merge 6 picture tags into 3')`

**Review by buuhuu — APPROVED** [review]

**Comment by anagarwa** [comment]
> image

@sdmcraft concerns should be resolved now

File: component-definition.json [hunk 3]
```diff
@@ -50,8 +50,9 @@
                 "template": {
                   "name": "Image Slide",
                   "model": "image-slide",
-                  "image": "/content/dam/aem-xwalk-test-harsh/content-at-scale.png",
-                  "imageAlt": "Sample Image"
+                  "image_desktop": "/content/dam/aem-xwalk-test-harsh/content-at-scale.png",
```
Comment (sdmcraft) [inline component-definition.json:53]: Are these defaults? If the expectation is to have some default assets, let's create a placeholder assets on DAM and place all such assets in it and document in README that the front end code expects such a folder to be present in DAM.

Comment (anagarwa) [reply component-definition.json]: @sdmcraft these were not necessary, I have removed it .
Thanks for the review it

**Pull request merged**